algosdk/source_map.py

```diff
@@ -0,0 +1,93 @@
+from typing import Dict, Any, List, Tuple
+
+
+class SourceMap:
+    """
+    Decodes a VLQ-encoded source mapping between PC values and TEAL source code lines.
+    Spec available here: https://sourcemaps.info/spec.html
+
+    Args:
+        source_map (dict(str, Any)): source map JSON from algod
+        delimiter (str, optional): delimiter for mappings
+    """
+
+    def __init__(self, source_map: Dict[str, Any], delimiter: str = ";"):
+        self.delimter = delimiter
+
+        self.version: int = source_map["version"]
+        self.sources: List[str] = source_map["sources"]
+        self.mapping: str = source_map["mapping"]
```

Comment:
@barnjamin Open to discussion:  While reviewing, I realized go-algorand incidentally returns _mapping_ instead of the spec defined _mappings_ key.

I think it's most prudent to address the underlying issue (which will cause a backwards incompatibility) because the current implementation deviates from the spec.

I apologize for the inconvenience - I reviewed the PR in-question and missed the deviation (https://github.com/algorand/go-algorand/pull/3726).

Reply:
Uhoh, good catch

Reply:
I can update the SDKs and tests to `mappings` but I suppose we should wait until the changes are merged into go-algorand before releasing? I can also put something in here to check for either key for now?

Reply:
@barnjamin Here's my thinking - What do you think?
* In go-algorand, the principled change is to:
  * Deprecate `mapping` via warning in OAS _description_, and continue populating it.  Circle back to removing it another day.
  * Add `mappings` field.  
*  In SDKs, check `mappings` first and if the key is absent, fallback to `mapping`.  When removing `mapping` server-side, we can remove the secondary look up in SDKs.

If the thinking makes sense, I can draft a story to track removal of the to-be-deprecated `mapping` field.

Reply:
@barnjamin Resolving - Leaving a few notes for context.

* Thanks for making changes to support falling back between `mapping` and `mappings`.  
  * I have slight preference to prefer the non-deprecated field (`mappings`).  But, it's a minor concern and preferring `mapping` works for me.
* I opened a few stories to track deprecation:
  * Epic covering all deprecations - https://github.com/algorand/algorand-sdk-testing/issues/200
  * Story for `mapping` removal - https://github.com/algorand/algorand-sdk-testing/issues/202